backend/app/modules/scheduler/MonitoringActor.scala

```diff
@@ -0,0 +1,69 @@
+package modules.scheduler
+
+import akka.actor.{Actor, Props}
+import models.Query
+
+import scala.collection.mutable
+
+object MonitoringActor {
+
+  /**
+    * Create a new instance of monitoring actor
+    */
+  def props = Props[MonitoringActor]
+
+  /**
+    * Message definition
+    */
+  case class StartProcess(query: String)
+  case class UpdateProcess(id: String, status: String)
+
+  case class ListProcess(status: Option[String] = None)
+}
+
+class MonitoringActor extends Actor {
+
+  /**
+    * Implicit import from companion object
+    */
+  import MonitoringActor._
+
+  /**
+    * List of all the process run by the server (including finished ones)
+    */
+  val processList = mutable.MutableList.empty[Query]
```

Comment:
Isn't this possibly dangerous? I mean if the number of query grow, the list of process will also grow a lot and eventually consume too much memory.